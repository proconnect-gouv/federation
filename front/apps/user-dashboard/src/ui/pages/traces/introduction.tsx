import classnames from 'classnames';
import { createUseStyles } from 'react-jss';

const useStyles = createUseStyles({
  introduction: {
    backgroundColor: '#FFFFFF', // accessibility purpose
  },
});

const IntroductionComponent = () => {
  const classes = useStyles();
  return (
    <div className={classnames(classes.introduction)}>
      <h4 className="is-blue-france mb8">
        <b>Votre historique de connexion</b>
      </h4>
      <p className="is-normal fr-text">
        Retrouver toutes les connexions et échanges de données effectués via
        FranceConnect ces six derniers mois. Cliquez sur une connexion pour en
        afficher les détails.
      </p>
    </div>
  );
};

IntroductionComponent.displayName = 'IntroductionComponent';

export default IntroductionComponent;
