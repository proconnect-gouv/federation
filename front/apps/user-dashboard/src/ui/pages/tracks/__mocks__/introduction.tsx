export const IntroductionComponent = jest.fn(() => <div>FooBar IntroductionComponent</div>);
