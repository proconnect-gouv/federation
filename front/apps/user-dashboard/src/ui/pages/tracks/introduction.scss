@import '@fc/dsfr/styles.scss';

.introduction {
  background-color: $white; // accessibility purpose
}
