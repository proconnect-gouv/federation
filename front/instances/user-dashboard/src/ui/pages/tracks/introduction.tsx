import React from 'react';

export const IntroductionComponent = React.memo(() => (
  <div>
    <h1 className="fr-h3 is-blue-france fr-mb-1w">Mon historique de connexion</h1>
    <p>
      Retrouvez toutes les connexions et échanges de données effectués via FranceConnect ces six
      derniers mois.
      <br />
      Cliquez sur une connexion pour en afficher les détails.
    </p>
  </div>
));

IntroductionComponent.displayName = 'IntroductionComponent';
