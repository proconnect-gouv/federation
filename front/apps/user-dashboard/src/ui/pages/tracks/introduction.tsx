import './introduction.scss';

export const IntroductionComponent = () => {
  return (
    <div className="introduction">
      <h4 className="is-blue-france mb8">
        <b>Votre historique de connexion</b>
      </h4>
      <p className="is-normal fr-text">
        Retrouver toutes les connexions et échanges de données effectués via
        FranceConnect ces six derniers mois. Cliquez sur une connexion pour en
        afficher les détails.
      </p>
    </div>
  );
};

IntroductionComponent.displayName = 'IntroductionComponent';
