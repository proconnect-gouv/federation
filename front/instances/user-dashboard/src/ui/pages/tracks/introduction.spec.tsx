import { render } from '@testing-library/react';

import { IntroductionComponent } from './introduction';

describe('IntroductionComponent', () => {
  it('should match the snapshot', () => {
    // when
    const { container } = render(<IntroductionComponent />);

    // then
    expect(container).toMatchSnapshot();
  });
});
