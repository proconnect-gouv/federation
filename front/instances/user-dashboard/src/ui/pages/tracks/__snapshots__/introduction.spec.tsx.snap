// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IntroductionComponent should match the snapshot 1`] = `
<div>
  <div>
    <h1
      class="fr-h3 is-blue-france fr-mb-1w"
    >
      <b>
        Votre historique de connexion
      </b>
    </h1>
    <p>
      Retrouver toutes les connexions et échanges de données effectués via FranceConnect ces six derniers mois.
      <br />
      Cliquez sur une connexion pour en afficher les détails.
    </p>
  </div>
</div>
`;
